tests/integration_tests/test_dask.py

```diff
@@ -141,3 +120,26 @@ def test_study_direction_best_value(client: Client, direction: str) -> None:
         expected = trials_value.min()
 
     np.testing.assert_allclose(expected, study.best_value)
+
+
+if _imports.is_successful():
+
+    @gen_cluster(client=True)
+    async def test_daskstorage_registers_extension(
+        c: "Client", s: "Scheduler", a: "Worker", b: "Worker"
+    ) -> None:
+        assert "optuna" not in s.extensions
+        await DaskStorage()
+        assert "optuna" in s.extensions
+        assert type(s.extensions["optuna"]) is _OptunaSchedulerExtension
+
+    @gen_cluster(client=True)
+    async def test_name(c: "Client", s: "Scheduler", a: "Worker", b: "Worker") -> None:
+        await DaskStorage(name="foo")
+        ext = s.extensions["optuna"]
+        assert len(ext.storages) == 1
+        assert type(ext.storages["foo"]) is optuna.storages.InMemoryStorage
+
+        await DaskStorage(name="bar")
+        assert len(ext.storages) == 2
+        assert type(ext.storages["bar"]) is optuna.storages.InMemoryStorage
```

Comment:
To avoid evaluate `gen_cluster` decorator, here I moved tests inside the if statement. There might be more straightforward solution. Perhaps we should merge this PR first, and then refactor this PR as a follow-up.